### PR TITLE
aws-lambda: mutual TLS authentication structure

### PR DESCRIPTION
**Pull request by kiesel**
This adds the structure as documented:

https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html

for the client certificate TLS authentication of the HTTP API of AWS APIGateway.

### CODE REVIEW

File: types/aws-lambda/trigger/api-gateway-proxy.d.ts [hunk 1]
```diff
@@ -138,6 +138,19 @@ export interface APIGatewayProxyEventV2 {
                 scopes: string[];
             };
         } | undefined;
+        authentication?: {
+            clientCert: {
+                clientCertPem: string;
+                issuerDN: string;
+                // not documented, so omitting:
+                // serialNumber: string;
```
Comment (savnik) [inline types/aws-lambda/trigger/api-gateway-proxy.d.ts:146]: Serial number is used as a unique identifier of the certificate from the CA in a general context of TLS. I see that AWS do not clearly state this - however i think that we can include this safely.
```suggestion
                serialNumber: string;
```

**Review by savnik — APPROVED** [review]
Everything looks good. One small comment which in my mind is not blocking but simply a suggestion.

File: types/aws-lambda/trigger/api-gateway-proxy.d.ts (hunk 1 above)
Comment (kiesel) [reply types/aws-lambda/trigger/api-gateway-proxy.d.ts]: Ok, I have included it!

While doing so I stumbled over a type that already exactly matches the clientCert structure - `APIGatewayEventClientCertificate` from `common/` - so I used it and adapted the tests accordingly.

**Review by savnik — APPROVED** [review]
LGTM

**Comment by kiesel** [comment]
Ready to merge

**Pull request merged**